Libraries/LibWeb/Painting/BorderPainting.cpp

```diff
@@ -426,7 +426,10 @@ void paint_border(DisplayListRecorder& painter, BorderEdge edge, DevicePixelRect
             points.append(Gfx::FloatPoint(rect.top_left().to_type<int>()));
         } else {
             Gfx::FloatPoint inner_right_angle_offset = Gfx::FloatPoint(0, joined_border_width.value() - radius.horizontal_radius);
-            points.append(Gfx::FloatPoint(rect.top_left().to_type<int>()) + inner_right_angle_offset);
+            auto top_left_point = Gfx::FloatPoint(rect.top_left().to_type<int>()) + inner_right_angle_offset;
+            // Clamp the final point to stay within the rect bounds to prevent rendering artifacts
+            top_left_point.set_y(AK::clamp(top_left_point.y(), rect.top().value(), rect.bottom().value()));
+            points.append(top_left_point);
```

Comment:
Why is all this just happening for `BorderEdge::Right`? Does the issue not exist for other borders?